scripts/bash/syncDotEnvWithParameterStore.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+# Stores every parameter in the .env file of --package-name on parameter store based on --environment
+# --package-name      The name of the package to store the .env variables of
+# --environment       The subpath to store the parameters under in parameter store, e.g. dev
+# --upload/--download Whether to save the .env state to parameter store, or download the parameters
+#                     from the store into the .env (defaults to upload)
+
+# Get current directory
+DIR=$(dirname "$0")
+
+# Set defaults
+DIRECTION="upload"
+
+# Getting command line arguments, taken from https://stackoverflow.com/questions/192249/how-do-i-parse-command-line-arguments-in-bash
+POSITIONAL=()
+while [[ $# -gt 0 ]]; do
+    key="$1"
+
+    case $key in
+    --package-name)
+        PACKAGE_NAME="$2"
+        shift # past argument
+        shift # past value
+        ;;
+    --environment)
+        ENVIRONMENT="$2"
+        shift # past argument
+        shift # past value
+        ;;
+    --path-to-package)
+        PATH_TO_PACKAGE="$2"
+        shift # past argument
+        shift # past value
+        ;;
+    --upload)
+        DIRECTION="upload"
+        shift # past argument
+        shift # past value
+        ;;
+    --download)
+        DIRECTION="download"
+        shift # past argument
+        shift # past value
+        ;;
+    esac
+done
+set -- "${POSITIONAL[@]}" # restore positional parameters
+
+if [[ -z $PACKAGE_NAME || -z $ENVIRONMENT ]]; then
+    echo "Please specify both --package-name and --environment (and optionally, --path-to-package, --upload, or --download)"
+    exit 0
+fi
+PARAMETER_STORE_PATH_PREFIX="/${PACKAGE_NAME}/${ENVIRONMENT}"
+
+if [[ -z $PATH_TO_PACKAGE ]]; then
+    # Use default path to package, based on ubuntu server setup
+    PATH_TO_PACKAGE="/home/ubuntu/tupaia/packages/${PACKAGE_NAME}"
+fi
+
+if [[ $DIRECTION == "upload" ]]; then
+    ${DIR}/storeDotEnvOnParameterStore.sh ${PATH_TO_PACKAGE} ${PARAMETER_STORE_PATH_PREFIX}
+else
+    ${DIR}/fetchDotEnvFromParameterStore.sh ${PATH_TO_PACKAGE} ${PARAMETER_STORE_PATH_PREFIX}
+fi
```

Comment:
💯

Reply:
I should have left a comment about motivation here. When we need to update an env var, we will need to change that variable, but also make sure that all the variables in the .env we're uploading contain `[branch-name]` rather than `dev`, so that the replacement process still works. I figured this would be a bit of a burden to do manually, and could result in mistakes, so I added this script that will pull down the env vars in their pure form as stored on SSM, without running the replacement regex. This means the steps to update become: pull from ssm, update _that_ file, push to ssm.

Reply:
Very good call! Perhaps we can add a script that does all 3 steps

Reply:
Oh yeah that could be cool...and then also makes the replacements again at the end (which is a weakness of this as it stands)